pkg/parquetquery/iters.go

```diff
@@ -252,6 +296,370 @@ func columnIteratorResultPoolPut(r *IteratorResult) {
 	}
 }
 
+// SyncIterator is like ColumnIterator but synchronous. It scans through the given row
+// groups and column, and applies the optional predicate to each chunk, page, and value.
+// Results are read by calling Next() until it returns nil.
+type SyncIterator struct {
+	// Config
+	column     int
+	columnName string
+	rgs        []pq.RowGroup
+	rgsMin     []RowNumber
+	rgsMax     []RowNumber // Exclusive, row number of next one past the row group
+	readSize   int
+	selectAs   string
+	filter     Predicate
+
+	// Status
+	curr            RowNumber
+	currRowGroup    pq.RowGroup
+	currRowGroupMin RowNumber
+	currRowGroupMax RowNumber
+	currChunk       pq.ColumnChunk
+	currPages       pq.Pages
+	currPage        pq.Page
+	currPageMax     RowNumber
+	currValues      pq.ValueReader
+	currBuf         []pq.Value
+	currBufN        int
+}
+
+var _ Iterator = (*SyncIterator)(nil)
+
+func NewSyncIterator(ctx context.Context, rgs []pq.RowGroup, column int, columnName string, readSize int, filter Predicate, selectAs string) *SyncIterator {
+
```

Comment:
ctx not used? it looks like we're losing the `columnIterator.iterate` span with this change which i find somewhat useful.

Reply:
oof, good point. our traces need attention generally and this PR is too obvious a win to block.

fine with the "NewSyncIterator" approach to see how it goes.

Reply:
👍 Tested it out and it works well enough for a first pass or if we have a better idea.  The side-effect is that the span starts as soon as the iter is created, instead of on the first "pull" like the async one.